game-core/src/main/java/games/strategy/engine/data/gameparser/LegacyPropertyMapper.java

```diff
@@ -55,7 +55,9 @@ public String mapPropertyName(final String propertyName) {
     return propertyName;
   }
 
-  static boolean ignoreOptionName(final String name) {
-    return name.equalsIgnoreCase("takeUnitControl") || name.equalsIgnoreCase("giveUnitControl");
+  static boolean ignoreOptionName(final String name, final String value) {
+    return name.equalsIgnoreCase("takeUnitControl")
+        || (name.equalsIgnoreCase("giveUnitControl")
+            && (value.equals("false") || value.equals("true")));
```

Comment:
Are you positive that `value` could never be null here? I don't remember if we translate empty values to empty strings or null values.

Reply:
`value` is never checked for null in the caller before using (see line 908 in GameParser) so I don't think it would ever be null.

Reply:
Considering we just had a NPE in `GameParser`, I would not presume it to be already correct. Another consideration, the 'value' is very context dependent, a DTD previously would have enforced values being provided, that is not necessarily guaranteed now. 

I wonder if there is any type of test coverage around this at all that could/should be updated.

Reply:
Hmm, I may be wrong about the DTD previously requiring a specific value, I'm not sure if it can constrain attribute values vs just requiring them being there. Regardless, it is context dependent if we can expect a value.

Reply:
Good point on the NPE.  It will now skip `giveUnitControl` if the value is `null` as well as `true` or `false`.

There is some test coverage but it requires creating a brand new xml with these attributes in it.  I'm looking at if there is a way to be able to create the xml in memory instead of committing the xml as a resource.

Reply:
If we can get away from XMLs for testing, that would be excellent.

The decoupling of XML parsing from the semantic data parsing into GameData was meant to help with this.

One should be able to set up a `org.triplea.map.data.elements.Game` object and then pass that to GameParser and avoid having to set up an XML. The former is well tested for a variety of XML snippets, so we can be pretty sure how that works. It should have some existing tests that might be interesting to know if we null out empty values. Beyond that, sending `GameParser` a defined `Game` object hopefully can do the trick. `GameParser` is still very rough, so it might take some plumbing work to accept a `Game` object here. Let me know if that turns out to be the case, for now a manual verification is enough or you can go the extra mile if you feel it's worthwhile given the context acquisition cost you will have payed.